src/TypeScript/NodeJSLocator/Locators/GlobalPathNodeLocator.cs

```diff
@@ -0,0 +1,108 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using SonarLint.VisualStudio.Integration;
+
+namespace SonarLint.VisualStudio.TypeScript.NodeJSLocator.Locators
+{
+    internal class GlobalPathNodeLocator : INodeLocator
+    {
+        private const string FileName = "node.exe";
+
+        private readonly IFileSystem fileSystem;
+        private readonly ILogger logger;
+
+        public GlobalPathNodeLocator(ILogger logger)
+            : this(new FileSystem(), logger)
+        {
+        }
+
+        internal GlobalPathNodeLocator(IFileSystem fileSystem, ILogger logger)
+        {
+            this.fileSystem = fileSystem;
+            this.logger = logger;
+        }
+
+        public string Locate()
+        {
+            var nodeExePath = GetPathToNodeExecutableFromEnvironment();
+
+            if (string.IsNullOrEmpty(nodeExePath))
+            {
+                logger.WriteLine(Resources.INFO_NotFoundInGlobalPath);
+                return null;
+            }
+
+            logger.WriteLine(Resources.INFO_FoundInGlobalPath, nodeExePath);
+            return nodeExePath;
+        }
+
+        /// <summary>
+        /// Copied from MS nodejstools repo: https://github.com/microsoft/nodejstools/blob/275e85d5cd95cad9122f59a76f9e49bead66101b/Nodejs/Product/Nodejs/Nodejs.cs#L112
+        /// </summary>
+        /// <remarks>
+        /// Copied as-is, except replacing `File.Exists` with <see cref="fileSystem.File.Exists"/> for testability
```

Comment:
What does the MS license say for the tools say about reusing the code? Do we need to acknowledge the original source?

Reply:
They have the generic apache license https://github.com/Microsoft/nodejstools/blob/main/LICENSE
but it's not filled out ("Copyright {yyyy} {name of copyright owner}"...).
I'll update our third-party notices and just point to the repo. I don't think there's any point in copy-pasting the generic apache. E.g.
```
License notice for MS NodeJsTools node.exe discovery code
---------------------------------------------------------
https://github.com/microsoft/nodejstools/blob/275e85d5cd95cad9122f59a76f9e49bead66101b/LICENSE
```